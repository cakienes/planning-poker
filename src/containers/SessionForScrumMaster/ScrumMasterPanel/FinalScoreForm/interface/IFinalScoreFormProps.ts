import IUserStory from '../../../../../interfaces/IUserStory';

export default interface IFinalScoreFormProps {
    handleSubmit?: (values: any) => void;
    activeUserStory?: IUserStory;
}
