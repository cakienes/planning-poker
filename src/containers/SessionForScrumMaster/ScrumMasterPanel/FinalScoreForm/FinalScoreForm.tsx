import React from 'react';
import { Field, reduxForm } from 'redux-form';
import Button from '../../../../components/Button/Button';
import FormFields from '../../../../components/FormFields/FormFields';
import { onlyNumber, required } from '../../../../helper/Validation';
import IFinalScoreFormProps from './interface/IFinalScoreFormProps';

export const FinalScoreForm: React.FC<IFinalScoreFormProps> = ({ handleSubmit, activeUserStory }) => {
    return (
        <form onSubmit={handleSubmit}>
            <div className="row justify-content-center">
                <div className="col-8 ">
                    <Field name="finalScore" component={FormFields} type="number" validate={[required, onlyNumber]} />
                </div>
            </div>
            <div className="row">
                <div className="col">
                    <Button label={`End Voting For ${activeUserStory && activeUserStory.storyName}`} type="submit" />
                </div>
            </div>
        </form>
    );
};

export default reduxForm<{}, any>({
    form: 'finalScoreForm',
})(FinalScoreForm);
