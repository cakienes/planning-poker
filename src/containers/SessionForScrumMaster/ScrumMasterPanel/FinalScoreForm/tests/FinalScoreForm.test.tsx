import React from 'react';
import { createRenderer, ShallowRenderer } from 'react-test-renderer/shallow';
import { FinalScoreForm } from '../FinalScoreForm';
import IFinalScoreFormProps from '../interface/IFinalScoreFormProps';

describe('FinalScoreForm.tsx', () => {
    let renderer: ShallowRenderer;
    const handleSubmitMock = jest.fn();

    let finalScoreFormProps: IFinalScoreFormProps = {
        handleSubmit: handleSubmitMock,
    };

    beforeEach(() => {
        renderer = createRenderer();
        finalScoreFormProps = {
            handleSubmit: handleSubmitMock,
            activeUserStory: undefined,
        };
    });

    it('renders correctly FinalScoreForm', async () => {
        expect(renderer.render(<FinalScoreForm {...finalScoreFormProps} />)).toMatchSnapshot();
    });

    it('renders correctly FinalScoreForm while activeUserStory is define', async () => {
        finalScoreFormProps = {
            ...finalScoreFormProps,
            activeUserStory: {
                storyName: 'storyName',
            },
        };
        expect(renderer.render(<FinalScoreForm {...finalScoreFormProps} />)).toMatchSnapshot();
    });
});
