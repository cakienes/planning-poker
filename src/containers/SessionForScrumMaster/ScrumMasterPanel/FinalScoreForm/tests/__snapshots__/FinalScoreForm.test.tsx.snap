// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FinalScoreForm.tsx renders correctly FinalScoreForm 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <div
    className="row justify-content-center"
  >
    <div
      className="col-8 "
    >
      <ForwardRef
        component={[Function]}
        name="finalScore"
        type="number"
        validate={
          Array [
            [Function],
            [Function],
          ]
        }
      />
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <Button
        label="End Voting For undefined"
        type="submit"
      />
    </div>
  </div>
</form>
`;

exports[`FinalScoreForm.tsx renders correctly FinalScoreForm while activeUserStory is define 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <div
    className="row justify-content-center"
  >
    <div
      className="col-8 "
    >
      <ForwardRef
        component={[Function]}
        name="finalScore"
        type="number"
        validate={
          Array [
            [Function],
            [Function],
          ]
        }
      />
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <Button
        label="End Voting For storyName"
        type="submit"
      />
    </div>
  </div>
</form>
`;
